modules/contact/include/constraints/ComputeFrictionalForceLMMechanicalContact.h

```diff
@@ -57,12 +65,29 @@ class ComputeFrictionalForceLMMechanicalContact : public ComputeWeightedGapLMMec
   /// A map from node to weighted gap
   std::unordered_map<const DofObject *, ADReal> _dof_to_weighted_tangential_velocity;
 
+  /// A map from node to weighted gap for an additional tangential direction (three dimensions)
+  std::unordered_map<const DofObject *, ADReal> _dof_to_weighted_tangential_velocity_dir;
```

Comment:
same question here. You're talking about gaps in the doc string but the data member is tangential velocity. Also does it make sense to have the original data member be a `std::unordered_map<const DofObject *, std::vector<ADReal>` and not have this additional new data member?

Reply:
Doc strings need to be revised throughout. Yeah, will update the data member to avoid new member.